irohad/ametsuchi/block_query.hpp

```diff
@@ -23,27 +23,28 @@
 #include <rxcpp/rx-observable.hpp>
 
 #include "common/types.hpp"
+#include "interfaces/iroha_internal/block.hpp"
+#include "interfaces/transaction.hpp"
 
 namespace iroha {
 
-  namespace model {
-    struct Transaction;
-    struct Block;
-  }
-
   namespace ametsuchi {
     /**
      * Public interface for queries on blocks and transactions
      */
     class BlockQuery {
      public:
+      using wTransaction =
```

Comment:
I think we can organise  a separate type of branches for refactoring, something like #1002 
Due to many collisions, I suppose it would be much easier if everyone just defines a `protected using` in his own "work field", and then to choose somebody who will move it to another file

Reply:
@lebdron what do you think?

Reply:
It seems like a good solution, because there is no central dependency like a single header file. In long term this required a better analysis, so it definitely works well in short term.